cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -204,6 +206,144 @@ func AvailablePackageSummaryFromChart(chart *models.Chart) (*corev1.AvailablePac
+		chart, err = s.manager.GetChartVersion(namespace, unescapedChartID, version)
+	}
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Unable to retrieve chart: %v", err)
+	}
+
+	availablePackageDetail, err := AvailablePackageDetailFromChart(&chart)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Unable to parse chart to an availablePackageDetail: %v", err)
+	}
+
+	return &corev1.GetAvailablePackageDetailResponse{
+		AvailablePackageDetail: availablePackageDetail,
+	}, nil
+}
+
+func AvailablePackageDetailFromChart(chart *models.Chart) (*corev1.AvailablePackageDetail, error) {
+	pkg := &corev1.AvailablePackageDetail{}
+
+	if chart.Name == "" {
+		return nil, status.Errorf(codes.Internal, "required field .Name not found on helm package: %v", chart)
+	}
+	pkg.Name = chart.Name
+	pkg.DisplayName = chart.Name
+
+	if chart.Icon == "" {
+		return nil, status.Errorf(codes.Internal, "required field .Icon not found on helm package: %v", chart)
+	}
+	pkg.IconUrl = chart.Icon
+
+	if chart.Description == "" {
+		return nil, status.Errorf(codes.Internal, "required field .Description not found on helm package: %v", chart)
+	}
+	pkg.ShortDescription = chart.Description
+	pkg.LongDescription = chart.Description
+
+	if chart.Maintainers == nil {
+		return nil, status.Errorf(codes.Internal, "required field .Maintainers not found on helm package: %v", chart)
+	}
+	pkg.Maintainers = []*corev1.Maintainer{}
+	for _, maintainer := range chart.Maintainers {
+		m := &corev1.Maintainer{Name: maintainer.Name, Email: maintainer.Email}
+		pkg.Maintainers = append(pkg.Maintainers, m)
+	}
+
+	if chart.ID == "" {
+		return nil, status.Errorf(codes.Internal, "required field .ID not found on helm package: %v", chart)
+	}
+	if chart.Repo == nil || chart.Repo.Namespace == "" {
+		return nil, status.Errorf(codes.Internal, "required field .Repo.Namespace not found on helm package: %v", chart)
+	}
+	pkg.AvailablePackageRef = &corev1.AvailablePackageReference{
+		Context:    &corev1.Context{Namespace: chart.Repo.Namespace},
+		Identifier: chart.ID,
+		Plugin:     GetPluginDetail(),
+	}
+
+	// We assume that chart.ChartVersions[0] will always contain either: the latest version or the specified version
+	if chart.ChartVersions == nil || len(chart.ChartVersions) == 0 {
+		return nil, status.Errorf(codes.Internal, "required field .chart.ChartVersions[0] not found on helm package: %v", chart)
+	}
+
+	if chart.ChartVersions[0].Version == "" {
+		return nil, status.Errorf(codes.Internal, "required field .chart.ChartVersions[0].Version not found on helm package: %v", chart)
+	}
+	pkg.PkgVersion = chart.ChartVersions[0].Version
+
+	if chart.ChartVersions[0].AppVersion == "" {
+		return nil, status.Errorf(codes.Internal, "required field .chart.ChartVersions[0].AppVersion not found on helm package: %v", chart)
+	}
+	pkg.AppVersion = chart.ChartVersions[0].AppVersion
+
+	if chart.ChartVersions[0].Readme == "" {
```

Comment:
Readme file is optional at https://helm.sh/docs/topics/charts/#the-chart-file-structure